internal/sail/server/server.go

```diff
@@ -51,7 +53,25 @@ func (s SailServer) newRoom(w http.ResponseWriter, req *http.Request) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	room := NewRoom()
+	if req.Method != http.MethodPost {
+		http.Error(w, "must be POST request", http.StatusBadRequest)
+		return
+	}
+
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error reading request: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	var newRoomReq model.SailNewRoomRequest
+	err = json.Unmarshal(body, &newRoomReq)
```

Comment:
it'd be more idiomatic to do json.NewDecoder(req.Body).Decode(&newRoomReq)

and save yourself some error handling

Reply:
i never knew! thanks :)